### ISSUE TITLE
TORCH_CPP_LOG_LEVEL cannot enable VLOG

### ISSUE THREAD

**Issue by ppwwyyxx** [pytorch/pytorch#135135]
### 🐛 Describe the bug

When not using GLOG, the `VLOG` macro is implemented as a negative severity: 
https://github.com/pytorch/pytorch/blob/eb0fd17bc451b604b7d7ed7981c4197666f4dd6e/c10/util/logging_is_not_google_glog.h#L95-L97

The severity is then compared with `FLAGS_caffe2_log_level`:
https://github.com/pytorch/pytorch/blob/eb0fd17bc451b604b7d7ed7981c4197666f4dd6e/c10/util/Logging.cpp#L415-L418

However, the flag at runtime comes from the `TORCH_CPP_LOG_LEVEL` envvar, which only supports 4 pre-defined severity levels:
https://github.com/pytorch/pytorch/blob/eb0fd17bc451b604b7d7ed7981c4197666f4dd6e/c10/util/Logging.cpp#L506-L539

As a result, when GLOG is not built, there is no way to control whether to print VLOG messages or not. 

### Versions

github main

**Comment by malfet** [comment]
Adding triage review as nobody is looking at module: logging.. And I though we've switched to TORCH_LOG a while back, which makes me wonder if it'll be better to just eliminate use of VLOG

**Comment by ppwwyyxx** [comment]
When GLOG is built (it might be in fbcode?), looks like VLOG would function correctly.  

In general it's helpful to have more log levels, and when building with GLOG, using `GLOG_vmodule` to control per-file verbosity is very handy in debugging.

**Comment by mikaylagawarecki** [comment]
From triage
- Kurt once looked into directing CPP logs to python but it did not happen
- Do we have a unified answer for what primitives one should be using for logging? What do we do with logs in open source? 
- cyyever sent a PR recently to remove VLOG but it is being used internally so we can't remoe it



cc @ezyang @malfet to follow up

**Comment by ezyang** [comment]
Oh, why don't we just add a -1 setting to TORCH_CPP_LOG_LEVEL lol

**Comment by ppwwyyxx** [comment]
Yep. `VLOG(k)` maps to severity `-k`, so we actually want `TORCH_CPP_LOG_LEVEL=-k` not just `-1`.
